### DISCUSSION TITLE
QNAP CPU limit question

### DISCUSSION THREAD

**Discussion by nomadski** [lloesche/valheim-server-docker#512] [Q&A] [answered]
In the QNAP help section it mentions limiting CPU usage with **cpus: '0.70'** being used as 70% for one CPU.  My NAS has dual CPUs so would that line still reflect 70% usage on both CPU's or would I have to amend it somehow?  (I'll be setting a 40% cap I think anyway to test performance).

Am new to Docker / Container Station, and not very tech savvy so apologies for the dumb question.

**Comment by Shawak** [comment, marked as answer]
No, it would reflect 70% of a single cpu. To reflect 70% on two CPUs you would have to enter `1.4`.
See: https://docs.docker.com/compose/compose-file/compose-file-v3/#resources

Edit: There is also a good answer to that on stackoverflow: https://stackoverflow.com/a/37871814

**Reply by nomadski** [reply]
Thanks, so 1.1 would be 40% on two CPUs? I will probably allocate more but would want the NAS to be responsive for media sharing as well.

**Reply by Shawak** [reply]
`0.8` should be 40% on two CPUs since `2.0` is max and `1.0` is 50%.

**Reply by nomadski** [reply]
Thanks again Shawak. I know I'm being dumb here - does docker already know my cpu count? So I am just setting the limit of what resources it can use and it can differentiate between (as the doc says) 0.70 being 70% for one CPU, and (as you replied) 0.8 being 40% for 2 CPUs?

Thanks for your patience :)

**Reply by Shawak** [reply]
I think you can specify that aswel. I don't know what the defaults are but I think that depends on the host system (e.g. docker desktop has a specified default cpu count for docker but linux doesn't limit them by default afaik).
